tests/platform_tests/test_link_flap.py

```diff
@@ -62,24 +64,37 @@ def __build_test_candidates(self, dut, fanouthosts):
         return candidates
 
 
-    def run_link_flap_test(self, dut, fanouthosts):
+    def run_link_flap_test(self, request, dut, fanouthosts):
         self.ports_shutdown_by_test = set()
 
         candidates = self.__build_test_candidates(dut, fanouthosts)
         if not candidates:
             pytest.skip("Didn't find any port that is admin up and present in the connection graph")
 
-        try:
-            for dut_port, fanout, fanout_port in candidates:
-                self.__toggle_one_link(dut, dut_port, fanout, fanout_port)
-        finally:
-            logging.info("Restoring fanout switch ports that were shut down by test")
-            for fanout, fanout_port in self.ports_shutdown_by_test:
-                logging.debug("Restoring fanout switch {} port {} shut down by test".format(fanout.hostname, fanout_port))
-                fanout.no_shutdown(fanout_port)
+        normalized_level = [mark.args for mark in request.node.iter_markers(name="completeness_level")][0]
```

Comment:
I'm not sure that I understand why the mark is necessary if we're just fetching it from the request and checking it in the method. How is this different than having a global `normalized_level` variable that we can reference or making it a field of the `testbed` fixture or something similar?

Reply:
There might be a scope of improvement and simplification.
To explain the machinery - there are two markers being used here:

1. `supported_completeness_level` is used to advertise the supported levels. This marker is being used by the normalization logic to fetch the supported levels and normalize it against the specified level. This will happen during testcase setup for every testcase. I think this marker is still needed for the normalization to work during testcase setup.
2. `completeness_level` is basically the normalized level that a testcase will be run against. **This does NOT have to be a marker.** Will the global variable not cause problem if there are two or more testcases being executed simultaneously? Making a field within `testbed` fixture definitely sounds more reasonable than just global variable.

Reply:
You are right, Neetha. The normalization is done per testcase and testbed fixture itself will not be testcase specific. I think Danny is suggesting to add a field in the testbed fixture for every testcase during testcase setup step. It wouldn't be very different approach from just setting a global variable. I am open to suggestions.
My preference with marker was simply to stamp the testcase with its normalized level in a not-so-global variable. Hence a marker in the testcase's request data.
What are the concerns with a marker carrying this normalized value?

Reply:
Update - modified the code to use just one marker instead of two. The normalized completeness level is now added to the list of supported levels.